src/Windows/Avalonia.Win32/Interop/TaskBarList.cs

```diff
@@ -1,14 +1,37 @@
 ﻿using System;
 using System.Runtime.InteropServices;
+using System.Threading;
 using static Avalonia.Win32.Interop.UnmanagedMethods;
 
 namespace Avalonia.Win32.Interop
 {
     internal class TaskBarList
     {
         private static IntPtr s_taskBarList;
+        private static bool s_initialized;
+        private static object s_lock = new();
+
         private static HrInit? s_hrInitDelegate;
         private static MarkFullscreenWindow? s_markFullscreenWindowDelegate;
+        private static SetOverlayIcon? s_setOverlayIconDelegate;
+
+        private static unsafe IntPtr Init()
+        {
+            int result = CoCreateInstance(in ShellIds.TaskBarList, IntPtr.Zero, 1, in ShellIds.ITaskBarList2, out IntPtr instance);
+
+            var ptr = (ITaskBarList3VTable**)instance.ToPointer();
+
+            s_hrInitDelegate ??= Marshal.GetDelegateForFunctionPointer<HrInit>((*ptr)->HrInit);
```

Comment:
Oh, yeah, it should be moved to MicroCom IDL files. 
If you want, I can push these changes to your branch, and remove this manual COM interop. Otherwise, it can wait.

Reply:
Let's wait, I'd rather have these changes merged sooner.